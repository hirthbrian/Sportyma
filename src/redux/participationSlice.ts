import { createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import type { RootState } from "./store";

import trainig1 from "../../data/training_1.json";
import trainig2 from "../../data/training_2.json";

// Define a type for the slice state
interface ParticipationState {
  [key: number]: string | null;
}

// Define the initial state using that type
const initialState: ParticipationState = {
  1: trainig1.participation,
  2: trainig2.participation,
};

export const participationSlice = createSlice({
  name: "participation",
  // `createSlice` will infer the state type from the `initialState` argument
  initialState,
  reducers: {
    approve: (state, action: PayloadAction<number>) => {
      state[action.payload] = "PRESENT";
    },
    refuse: (state, action: PayloadAction<number>) => {
      state[action.payload] = "ABSENT";
    },
  },
});

export const { approve, refuse } = participationSlice.actions;

// Other code such as selectors can use the imported `RootState` type
export const selectParticipationStatus = (state: RootState) =>
  state.participation;

export default participationSlice.reducer;
