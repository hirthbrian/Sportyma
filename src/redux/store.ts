import { configureStore } from "@reduxjs/toolkit";
import participation from "./participationSlice";

export const store = configureStore({
  reducer: {
    participation,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
