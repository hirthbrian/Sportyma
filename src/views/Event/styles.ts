import styled from "styled-components/native";

import { Colors } from "../../types";

export const Container = styled.ScrollView`
  flex: 1;
  padding: 0 20px;
  background-color: ${Colors.White};
`;
