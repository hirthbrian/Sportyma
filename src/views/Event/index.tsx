import { useEffect } from "react";
import { useNavigation, useRoute } from "@react-navigation/native";

import {
  PageHeader,
  ParticipationStatus,
  EventDetails,
  Information,
} from "../../organisms";
import { Separator } from "../../atoms";

import { loadEventData } from "../../utils";

import { Container } from "./styles";

const Event = () => {
  const navigation = useNavigation();
  const route = useRoute();
  const data = loadEventData();
  const event = data[route.params.id - 1];

  const { id, title, startdate, enddate, location, information } = event;

  useEffect(() => {
    navigation.setOptions({ title });
  }, []);

  return (
    <Container>
      <PageHeader date={startdate} />
      <Separator />
      <ParticipationStatus eventId={id} />
      <Separator />
      <EventDetails
        title={title}
        startDate={startdate}
        endDate={enddate}
        location={location}
      />
      {information && (
        <>
          <Separator />
          <Information info={information} />
        </>
      )}
    </Container>
  );
};

export default Event;
