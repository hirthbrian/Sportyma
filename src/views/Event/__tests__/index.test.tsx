import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

import { store } from "../../../redux/store";

import Event from "..";

const mockedNavigate = jest.fn();
const mockedSetOptions = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
      setOptions: mockedSetOptions,
    }),
    useRoute: () => ({
      params: {
        id: 1,
      },
    }),
  };
});

describe("<Event />", () => {
  let props;

  beforeEach(() => {
    props = {};
  });

  it("renders correcly", () => {
    expect(
      renderer
        .create(
          <Provider store={store}>
            <Event {...props} />
          </Provider>
        )
        .toJSON()
    ).toBeTruthy();
  });
  it("match snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Event {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
