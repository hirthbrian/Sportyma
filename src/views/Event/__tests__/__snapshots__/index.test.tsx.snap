// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> match snapshot 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 0,
      },
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "paddingTop": 15,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#2559f2",
              "borderRadius": 30,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Kanit-Regular",
                "fontSize": 16,
                "letterSpacing": 1,
                "paddingBottom": 4,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 4,
              },
              Object {
                "fontSize": 12,
              },
            ]
          }
        >
          16 FEBRUARY
        </Text>
      </View>
      <Text
        size={20}
        style={
          Array [
            Object {
              "color": "#152D4B",
              "fontFamily": "Kanit-Regular",
              "fontSize": 20,
            },
          ]
        }
      >
        21:35
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#dfe5ec",
            "height": 1,
            "lineHeight": 28,
            "marginBottom": 15,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 15,
          },
        ]
      }
    />
    <Text
      size={20}
      style={
        Array [
          Object {
            "color": "#152D4B",
            "fontFamily": "Kanit-Medium",
            "fontSize": 20,
          },
        ]
      }
      weight={500}
    >
      Ma participation
    </Text>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingTop": 15,
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            color="#29D48C"
            isActive={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#2559f233",
                  "borderRadius": 50,
                  "borderWidth": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Image
              color="#29D48C"
              isActive={false}
              source={1}
              style={
                Array [
                  Object {
                    "height": 18,
                    "marginRight": 10,
                    "tintColor": "#2559F2",
                    "width": 18,
                  },
                ]
              }
            />
            <Text
              isActive={false}
              size={13}
              style={
                Array [
                  Object {
                    "color": "#2559F2",
                    "fontFamily": "Kanit-Medium",
                    "fontSize": 13,
                    "letterSpacing": 1,
                  },
                ]
              }
              weight={500}
            >
              PRÉSENT
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "width": 15,
            },
          ]
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <View
            color="#EF4948"
            isActive={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#2559f233",
                  "borderRadius": 50,
                  "borderWidth": 2,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Image
              color="#EF4948"
              isActive={false}
              source={1}
              style={
                Array [
                  Object {
                    "height": 18,
                    "marginRight": 10,
                    "tintColor": "#2559F2",
                    "width": 18,
                  },
                ]
              }
            />
            <Text
              isActive={false}
              size={13}
              style={
                Array [
                  Object {
                    "color": "#2559F2",
                    "fontFamily": "Kanit-Medium",
                    "fontSize": 13,
                    "letterSpacing": 1,
                  },
                ]
              }
              weight={500}
            >
              ABSENT
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#dfe5ec",
            "height": 1,
            "lineHeight": 28,
            "marginBottom": 15,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 15,
          },
        ]
      }
    />
    <Text
      size={20}
      style={
        Array [
          Object {
            "color": "#152D4B",
            "fontFamily": "Kanit-Medium",
            "fontSize": 20,
          },
        ]
      }
      weight={500}
    >
      Entraînement à Strasbourg
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingTop": 15,
          },
        ]
      }
    >
      <View
        color="#E3E8F9"
        size={36}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#E3E8F9",
              "borderRadius": 18,
              "height": 36,
              "justifyContent": "center",
              "marginRight": 10,
              "width": 36,
            },
          ]
        }
      >
        <Image
          color="#2559F2"
          source={1}
          style={
            Array [
              Object {
                "height": 18,
                "tintColor": "#2559F2",
                "width": 18,
              },
            ]
          }
        />
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#152D4B",
                "fontFamily": "Kanit-Regular",
                "fontSize": 16,
              },
            ]
          }
        >
          Thursday 16 February 2023
        </Text>
        <Text
          color="#778EB9"
          style={
            Array [
              Object {
                "color": "#778EB9",
                "fontFamily": "Kanit-Light",
                "fontSize": 16,
              },
            ]
          }
          weight={300}
        >
          21:35 - 21:35
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingTop": 15,
          },
        ]
      }
    >
      <View
        color="#E3E8F9"
        size={36}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#E3E8F9",
              "borderRadius": 18,
              "height": 36,
              "justifyContent": "center",
              "marginRight": 10,
              "width": 36,
            },
          ]
        }
      >
        <Image
          color="#2559F2"
          source={1}
          style={
            Array [
              Object {
                "height": 18,
                "tintColor": "#2559F2",
                "width": 18,
              },
            ]
          }
        />
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#152D4B",
                "fontFamily": "Kanit-Regular",
                "fontSize": 16,
              },
            ]
          }
        >
          Stade de la Meinau
        </Text>
        <Text
          color="#778EB9"
          style={
            Array [
              Object {
                "color": "#778EB9",
                "fontFamily": "Kanit-Light",
                "fontSize": 16,
              },
            ]
          }
          weight={300}
        >
          12 rue de l’Extenwoerth,
67100 Strasbourg
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#dfe5ec",
            "height": 1,
            "lineHeight": 28,
            "marginBottom": 15,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 15,
          },
        ]
      }
    />
    <Text
      size={20}
      style={
        Array [
          Object {
            "color": "#152D4B",
            "fontFamily": "Kanit-Medium",
            "fontSize": 20,
          },
        ]
      }
      weight={500}
    >
      Informations
    </Text>
    <Text
      color="#152D4B"
      style={
        Array [
          Object {
            "color": "#152D4B",
            "fontFamily": "Kanit-Light",
            "fontSize": 16,
          },
        ]
      }
      weight={300}
    >
      N’oubliez pas vos lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </Text>
  </View>
</RCTScrollView>
`;
